docs/developer/payments.mdx

```diff
@@ -197,8 +197,19 @@ The [`transactionInitialize`](api-reference/payments/mutations/transaction-initi
 - `paymentGateway.data` - The data to be passed to the payment gateway.
 - `amount` - The amount requested for the payment. If not provided, the difference between `checkout.total` and transactions that are already processed will be sent.
 - `action` - The expected action (charge or authorization). By default, the `channel.defaultTransactionFlowStrategy` will be used and can be configured from the Saleor Dashboard. The default value can be overridden only by an app that has `HANDLE_PAYMENTS` permission.
+- `idempotencyKey` - Idempotency key which will be passed to the payment app. The key needs to be unique for the pair - `idempotencyKey` and `paymentGateway.id`. This field was introduced in **Saleor 3.14**.
 - `customerIpAddress` - The customer's IP address. If not provided, Saleor will try to determine the customer's IP address on its own. The customer's IP address will be passed to the payment app. The IP should be in ipv4 or ipv6 format. The field can be used only by an app that has `HANDLE_PAYMENTS` permission. This field was introduced in **Saleor 3.16**.
 
+Depending on the provided `idempotencyKey` value, the below results will happen:
+
+- If `idempotencyKey` key is not provided, a new one will be generated automatically.
+- If `idempotencyKey` is provided as an empty string, the error with the INVALID code will be raised.
+- If provided pair: `idempotencyKey` and `paymentGateway.id` is already assigned to the existing transaction which belongs to a different order/checkout, the error with the UNIQUE code will be raised.
+- If provided pair: `idempotencyKey` and `paymentGateway.id` is used more than once for a single order/checkout but with different mutation input, the error with UNIQUE code will be raised.
+- If provided pair: `idempotencyKey` and `paymentGateway.id` is used more than once for a single order/checkout, with the same input data, the existing `TransactionItem` will be used to send the request to the payment app.
+
+The idempotency is used to detect the same payment request made by transactionInitialize. The idempotency key should be generated by the storefront and passed as a mutation input. Any duplicated transactionInitialize calls should provide the same idempotency key.
+
```

Comment:
Maybe we should provide an example on how to generate the idempotency key on storefront?

I think we should suggest using UUID. In JS, this can be done using [`crypto.randomUUID`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID).

This key should also be persisted in some form of storage. In JS, this can be done via `localStorage`

```suggestion
In Web, UUID can be generated by using Crytpo Web API:

`js
const idempotencyKey = window.crypto.randomUUID();
`

The key should be also stored in a form of persistant storage. This way it can be re-used even if the user leaves the page / app. In Web this can be done with `localStorage`:

`js
// set
localStorage.setItem("idempotencyKey", idempotencyKey);

// get
const idempotencyKey = localStorage.getItem("idempotencyKey");
`

```